packages/blocks/azure/src/lib/actions/azure-advisor-action.ts

```diff
@@ -0,0 +1,81 @@
+import { createAction, Property } from '@openops/blocks-framework';
+import { azureAuth } from '@openops/common';
+import { logger } from '@openops/server-shared';
+import { runCommand } from '../azure-cli';
+import { subDropdown, useHostSession } from '../common-properties';
+
+export const advisorAction = createAction({
+  auth: azureAuth,
+  name: 'advisor',
+  description: 'Get Azure Advisor Cost Recommendations',
+  displayName: 'Get Advisor Cost Recommendations',
+  props: {
+    useHostSession: useHostSession,
+    subscriptions: subDropdown,
+    resourceIds: Property.Array({
+      displayName: 'Resource IDs',
+      description:
+        'One or more resource IDs (space-delimited). If provided, "Resource Group" should not be specified.',
+      required: false,
+    }),
+    resourceGroup: Property.ShortText({
+      displayName: 'Resource Group',
+      description:
+        'Name of a resource group. . If provided, "Resource Ids" should not be specified.',
+      required: false,
+    }),
+  },
+  async run(context) {
+    const command = buildCommand(context.propsValue);
+    logger.info(`Running command: ${command}`);
+    try {
+      const result = await runCommand(
+        command,
+        context.auth,
+        context.propsValue.useHostSession?.['useHostSessionCheckbox'],
+        context.propsValue.subscriptions?.['subDropdown'],
+      );
+
+      try {
+        const jsonObject = JSON.parse(result);
+        return jsonObject;
+      } catch (error) {
+        return result;
+      }
+    } catch (error) {
+      logger.error('Failed to fetch Azure cost recommendations', {
+        command: command,
+        error: error,
+      });
+      let message =
+        'An error occurred while fetching Azure cost recommendations: ';
+      if (String(error).includes('login --service-principal')) {
```

Comment:
kinda ugly for now but I see your point, let's see once we get a third action that needs it